jib-core/src/main/java/com/google/cloud/tools/jib/filesystem/PermissionsHelper.java

```diff
@@ -62,13 +64,13 @@ public static int toPermissionBits(ImmutableSet<PosixFilePermission> permissions
   public static ImmutableSet<PosixFilePermission> toPermissionSet(int permissions) {
     Preconditions.checkArgument(
         permissions >= 0 && permissions <= 0777, "Permissions must be between 000 and 777 octal");
-    ImmutableSet.Builder<PosixFilePermission> permissionsSetBuilder = ImmutableSet.builder();
+    EnumSet<PosixFilePermission> permissionsSetBuilder = EnumSet.noneOf(PosixFilePermission.class);
     for (Entry<PosixFilePermission, Integer> entry : permissionMap.entrySet()) {
       if ((permissions & entry.getValue()) != 0) {
         permissionsSetBuilder.add(entry.getKey());
       }
     }
-    return permissionsSetBuilder.build();
+    return ImmutableSet.copyOf(permissionsSetBuilder);
```

Comment:
Why does this need to be immutable? It's not a reference that we hold on to or care about.  If the caller wants to modify it to add or remove permissions now we force them to copy it. And it should just be an `EnumSet`.